source/src/BNFC/Backend/Java/CFtoJLex15.hs

```diff
@@ -45,6 +45,13 @@ import BNFC.Utils (cstring)
 import BNFC.Backend.Common.NamedVariables
 import Text.PrettyPrint
 
+-- This is for now the workaround to differentiate JLex and JFlex.
```

Comment:
This is surely a comment I should have removed. 
I don't think it is *terribly* bad. I get nuisance from see the checks spreaded and having boolean arguments to functions.
I agree completely with your observation about having two different functions.
For your proposal I think it's OK leaving the code as it is, it is probably not worth the effort.
I beg your pardon for using such an inappropriate hyperbole.